lib/rules/no-constant-condition.js

```diff
@@ -16,6 +16,27 @@ module.exports = function(context) {
     // Helpers
     //--------------------------------------------------------------------------
 
+
+    /**
+     * Checks if a branch node of LogicalExpression short circuits the whole condition
+     * @param {ASTNode} node The branch of main condition which needs to be checked
+     * @param {string} operator The operator of the main LogicalExpression.
+     * @returns {boolean} true when condition short circuits whole condition
+     */
+    function isShortCircuit(node, operator) {
```

Comment:
I think the problem is that `a && false` is a logical identity but not a short circuit. If you only check `false && a`, that's a short circuit (and a logical identity).